### PR TITLE
Add user manual doc; organize docs better

### LINKED ISSUE #360: Improve/cleanup documentation
Our docs are just a smattering of examples right now. New features and fixes have been added without updating the docs. Comprehensive docs and examples for everything supported should be written. Ideally the format and order will reflect the Docker API docs here https://docs.docker.com/engine/reference/api/docker_remote_api_v1.18.

### PR DESCRIPTION
**Pull request by davidxia**
Fixes #360

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,16 @@
-# Docker Client [![Circle CI](https://circleci.com/gh/spotify/docker-client.png?style=badge)](https://circleci.com/gh/spotify/docker-client) [![codecov.io](https://codecov.io/github/spotify/docker-client/coverage.svg?branch=master)](https://codecov.io/github/spotify/docker-client?branch=master)
-
+# Docker Client [![Build Status](https://travis-ci.org/spotify/docker-client.svg?branch=master)](https://travis-ci.org/spotify/docker-client) [![codecov.io](https://codecov.io/github/spotify/docker-client/coverage.svg?branch=master)](https://codecov.io/github/spotify/docker-client?branch=master)
 
 This is a simple [Docker](https://github.com/docker/docker) client written in Java.
+We build and test docker-client on Docker API versions 1.6 - 1.10 (specifically the ones (here)[1]).
```
Comment (mattnworb) [inline README.md]: nitpick: the API versions have different values than the actual client/server version for some reason: https://docs.docker.com/engine/reference/api/docker_remote_api/

so we should refer to either "Docker version" or "Docker API version" and use the appropriate column in the table on that page

**Comment by mattnworb** [comment]
whoa, awesome job on the user manual

File: README.md (hunk 1 above)
Comment (davidxia) [inline README.md]: Ah, yes my mistake.

**Comment by davidxia** [comment]
Thanks. I just copied the order of Docker's API doc: https://docs.docker.com/engine/reference/api/docker_remote_api_v1.18/

**Comment by mattnworb** [comment]
:+1:

**Pull request merged**